jenkins/infra/terraform/main.tf

```diff
@@ -0,0 +1,312 @@
+provider "aws" {
+  region = "${var.region}"
+}
+
+terraform {
+  backend "s3" {
+    bucket = "mdn-state-4e366a3ac64d1b4022c8b5e35efbd288"
+    key    = "terraform/ci"
+    region = "us-west-2"
+  }
+}
+
+resource "aws_key_pair" "mdn" {
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  key_name   = "mdn"
+  public_key = "${var.ssh_pubkey}"
+}
+
+# Create a new load balancer
+resource "aws_elb" "ci" {
+  name    = "ci-elb-${var.project}"
+  subnets = ["${data.aws_subnet_ids.subnet_id.ids}"]
+
+  listener {
+    instance_port     = 80
+    instance_protocol = "http"
+    lb_port           = 80
```

Comment:
do we need port 80?

Reply:
Well sort of, I'm not listening on 443 or any other port in the backend just port 80 but on the backed I added the following to nginx:

```
  if ($http_x_forwarded_proto != 'https') {
    return 301 https://$host$request_uri;
  }
```

Which will then redirect back to the ELB on port 443 which terminates the SSL cert using an ACM cert

Reply:
I had the same question as @metadave. I tried connecting via HTTP and was permanently redirected to HTTPS, so @limed's nginx-based redirect is working. 👍